pimd/pim_cmd.c

```diff
@@ -7394,11 +7395,16 @@ DEFUN (no_ip_pim_rp,
 		 "frr-pim:pimd", "pim", vrfname, "frr-routing:ipv4",
 		 argv[idx_rp]->arg);
 
-	snprintf(group_list_xpath, sizeof(group_list_xpath), "%s/group-list",
-		 rp_xpath);
 
-	snprintf(group_xpath, sizeof(group_xpath), "%s[.='%s']",
-		 group_list_xpath, group_str);
+	printed = snprintf(group_list_xpath, sizeof(group_list_xpath),
+			   "%s/group-list", rp_xpath);
+
+	assert(printed <= (int)sizeof(group_list_xpath) && "Xpath too long");
```

Comment:
Please do NOT make this `assert`s, we're getting input from a user on a `vty` here, so that's where an error report should go, and not result in just crashing the daemon.

Reply:
I put asserts here to kind of force the issue, to make the point that this should *never* happen if `XPATH_MAXLEN` is actually the maximum length of an xpath.

Since the build's currently broken I can change it to vty error messages but I think this bears further consideration.

Reply:
Actually all of these have the comparison backwards too :joy:

Reply:
Regarding the assert/vty_out discussion – what do we expect users to do when they receive "Xpath too long" error? None of those commands has variable-length arguments (I don't count IP address as such), so there's actually nothing user can do in this case.
It's up to the developer to make sure that XPATH_MAX is enough so probably assert is better here.

Reply:
I missed this @idryzhov. I'm not too concerned with that, I'm only concerned at the moment with fixing the build in a not totally wrong way.